docs/BiasLanguageMigration.md

```diff
@@ -0,0 +1,111 @@
+
+## How to remove old CRs after migration?
+
+The script will not delete the older CRs even after the deployment is migrated. This allows you to validate and verify the environment once again before fully removing the CRs. 
+The script will save the CRs that can be removed in `to_remove_CRs`, you can remove them manually using `kubectl delete -f to_remove_CRs/<CR-Name>` after validation.
+
+## Execution Modes
+
+### Generate Mode
+
+Run the script in `tools/bias_lang_migration.sh` passing `generate` as an argument as well as the `namespace` of the current deployment.
+
+Example:
+```
+./tools/bias_lang_migration.sh generate default
+```
+
+This example will generate the yaml needed for migrating a deployment in the default namespace. 
+The script will generate the following folder structure:
+
+```
+current_folder/
+├── default.CR_migrations
+│    ├── backup  
+│    ├── original
+│    ├── updated
+│    ├── to_remove_CRs    
+```
+**backup:** Backup configs from the current deployment
+
+**original:** Stores All CRs as currently deployed
+
+**updated:** Stores the converted CRs to be used in the migration
+
+**to_remove_CRs:** Stores CRs that can be removed (old Masters) after the migration
+
+
+#### After the generation is completed
+The `updated` folder has all of the files needed for the migration. The next step is to apply each one of them following the oder below. This order ensures the correct order of Statefulset updates. 
+If any of these CRs are not deployed in your environment, you can skip to the next item.
+
+**Order for migration:**
+
+1) Apply the label biaslangmasternode=yes in the Node where LicenseMaster is deployed
+2) Apply the LicenseManager CR generated
+3) Apply LicenseManager Rsync Job for /etc PVC
+4) Apply LicenseManager Rsync Job for /var PVC
+5) Remove the label biaslangmasternode=yes
+6) Apply the label biaslangmasternode=yes in the Node where ClusterMaster is deployed
+7) Apply the ClusterManager CR generated
+8) Apply ClusterManager Rsync Job for /etc PVC
+9) Apply ClusterManager Rsync Job for /var PVC 
+10) Apply IndexerCluster CR
+11) Apply SearchHeadCluster CR
+12) Apply Standalone CR
+
+Once you validate your environment, you can remove the older CM and LM CRs saved in the folder `to_remove_CRs`
+
+For more details on how to apply labels check: [K8s docs](https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/)
+
+
+### Migrate Mode
+
+Run the script in `tools/bias_lang_migration.sh` passing "migrate" as an argument as well as the namespace of the current deployment.
+
+Example:
+```
+./tools/bias_lang_migration.sh migrate default
+```
+
+This will start a migration where we clone deployed ClusterMaster/LicenseMaster into a new independent ClusterManager/LicenseManager. Then we use a Rsync job to copy the data from Master to Manager.
+Next the script update the Statefulsets for all CRs that references the CM/LM originally deployed. 
```

Comment:
Next  script updates  originally deployed Statefulsets for all CRs that references the CM/LM.

Reply:
Done